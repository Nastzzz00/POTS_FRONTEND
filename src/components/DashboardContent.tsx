import { Button, Calendar, Card, Divider, Drawer, List, Typography } from 'antd'
import moment from 'moment'
import React, { Fragment, useState } from 'react'
import InfiniteScroll from 'react-infinite-scroller'
import Avatar from '../components/Avatar'
import { statusReport } from '../data/mockData'
import StatusItem from './StatusItem'
export interface IDashboardContentProps {
  list?: any
}

const DashboardContent = (props: IDashboardContentProps) => {
  let format = 'YYYY-MM-DD'
  const { list } = props
  const [state, setState] = useState({
    visible: false,
    selectedDate: moment().format(format),
  })
  const { Title, Text } = Typography

  const filterbydate = list.filter(
    (x: any) =>
      moment(x.documentDate).format('MMMM D, YYYY') ==
      moment(state.selectedDate).format('MMMM D, YYYY'),
  )

  const onChange = (value: any) => {
    setState({
      ...state,
      selectedDate: value.format(format),
    })
  }

  const showDrawer = () => {
    setState({
      ...state,
      visible: true,
    })
  }

  const onClose = () => {
    setState({
      ...state,
      visible: false,
    })
  }
  return (
    <Fragment>
      <div className="content1">
        <div className="status">
          <div className="avatar">
            <div className="rows">
              <div>
                <Avatar />
              </div>
            </div>
          </div>
          <div className="banner">
            <div style={{ marginLeft: 20 }}>
              <Title
                level={4}
                style={{ margin: 0, paddingTop: '5px', color: '#3182CE' }}>
                Welcome Back!
              </Title>
              <Text>Your last log-in was:</Text>
              <br></br>
              <Text>January 12, 2020 14:33 GST</Text>
              <br /> <br />
              <Text>Today is {moment().format()}</Text>
            </div>

            <Card
              style={{
                width: 300,
                borderColor: 'white',
                marginRight: 100,
              }}>
              <text style={{ marginLeft: 20 }}>
                Here are updates on our goodies
              </text>
              <div className="statusItem">
                <StatusItem status={statusReport} />
              </div>
            </Card>
          </div>
        </div>
        <div>
          <Divider orientation="left" dashed={true}>
            Order Calendar Summary
          </Divider>
        </div>

        <div className="calendar">
          <div className="calendar1">
            <Calendar fullscreen={false} onChange={onChange} />
          </div>
          <div className="list">
            <InfiniteScroll
              initialLoad={false}
              pageStart={0}
              loadMore={() => console.log('LOAD MORE')}
              useWindow={false}>
              <List
                itemLayout="horizontal"
                dataSource={filterbydate}
                renderItem={(item: any) => (
                  <List.Item>
                    <List.Item.Meta
                      title={<a> PO# {item.purchaseOrderNo}</a>}
                      description={
                        <div>
                          <Text>
                            Document Date:
                            {moment(item.documentDate).format('MMMM D, YYYY')}
                          </Text>
                        </div>
                      }
                    />

                    <List.Item>
                      <Button type="primary" onClick={showDrawer}>
                        View Order
                      </Button>
                    </List.Item>
                  </List.Item>
                )}
              />
            </InfiniteScroll>
          </div>
        </div>
      </div>
      <div>
        <div className="content2">
          <h5>To-Do List</h5>
          <Text>
            {'\u25A0'} Remind supplier to update <br />
            delivery status via phone call
          </Text>
          <br />
          <Text>
            {'\u25A0'} Create a presentation of <br />
            weekly delivery status
          </Text>
          <br />
          <Text delete>
            {'\u25A0'} Send email to partners <br />
          </Text>
          <br />
        </div>
        <div className="content3">
          <h5>Reminders</h5>
          <Text>
            {'\u25A0'} Meeting with OPs 10am,
            <br />
            March 3, 2020
          </Text>
          <br />
          <Text>
            {'\u25A0'} Innaguration of new ware- <br />
            house facility, March 10, 2020
          </Text>
        </div>
        <div className="content4">
          <Button type="link" icon="plus">
            <Text> Add widget</Text>
          </Button>
        </div>
      </div>

      <Drawer
        title="Purchase Order Details"
        width={640}
        placement="right"
        closable={false}
        onClose={onClose}
        visible={state.visible}>
        <p>Some Purchase Order Details...</p>
        <p>Some Purchase Order Details...</p>
        <p>Some Purchase Order Details...</p>
      </Drawer>
    </Fragment>
  )
}

export default DashboardContent
