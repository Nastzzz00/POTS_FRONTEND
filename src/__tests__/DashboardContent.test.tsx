/**
 * @format
 */

import React from 'react'
import { shallow } from 'enzyme'
// import ShallowRenderer from "react-test-renderer/shallow";

// // Note: test renderer must be required after react-native.
// import renderer from "react-test-renderer";

import DashboardContent from '../components/DashboardContent'

const props = {
  title: 'Dashboard Content',
}

describe('Dashboard Content', () => {
  test('renders correctly', () => {
    expect(
      shallow(<DashboardContent list={[{ test: 'test' }]} />),
    ).toMatchSnapshot()
  })
})
