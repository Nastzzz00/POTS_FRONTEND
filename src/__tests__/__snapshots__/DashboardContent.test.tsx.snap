// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Content renders correctly 1`] = `ShallowWrapper {}`;
